File: frontend/src/types/message.ts
export type ChatRoom = {
    chat_room_id: string;
    chat_room_name: string;
    chat_room_type: string;
};

export type ChatRoomMember = {
    chat_room_id: string;
    user_id: string;
};

export type Message = {
    message_id: string;
    chat_room_id: string;
    user_id: string;
    content: string;
    timestamp: number;
};

export type OpenConversations = {
    user_id: string;
    chat_room_id: string;
};

File: backend/service-message/models/message.go
package models

type ChatRoom struct {
	ChatRoomId   string `gorm:"primaryKey;"`
	ChatRoomName string `gorm:"not null;"`
	ChatRoomType string `gorm:"not null;"`
}

type ChatRoomMember struct {
	ChatRoomId string `gorm:"primaryKey;"`
	UserId     string `gorm:"primaryKey;"`
}

type Message struct {
	MessageId  string `gorm:"primaryKey;"`
	ChatRoomId string `gorm:"not null;"`
	UserId     string `gorm:"not null;"`
	Content    string `gorm:"not null;"`
	Timestamp  int64  `gorm:"not null;"`
}

type OpenConversations struct {
	UserId     string `gorm:"primaryKey;"`
	ChatRoomId string `gorm:"primaryKey;"`
}
